src/layer.js

```diff
@@ -133,6 +133,103 @@ export default class LayerClient {
     return response.json();
   }
 
+  /**
+   * Get all layers of a workspace.
+   *
+   * @param {String} workspace The workspace
+   *
+   * @throws Error if request fails
+   *
+   * @return {Object} An object with the information about the layers
+   */
+  async getLayers (workspace) {
+    const response = await fetch(this.url + 'workspaces/' + workspace + '/layers.json', {
+      credentials: 'include',
+      method: 'GET',
+      headers: {
+        Authorization: this.auth,
+      }
+    });
+
+    if (!response.ok) {
+      const geoServerResponse = await getGeoServerResponseText(response);
+      throw new GeoServerResponseError(null, geoServerResponse);
+    }
+
+    return await response.json();
+  }
+
+  /**
+   * Returns information about a cascaded WMS layer.
+   *
+   * @param {String} workspace The workspace
+   * @param {String} datastore The datastore
+   * @param {String} layerName The WMS layer name
+   *
+   * @throws Error if request fails
+   *
+   * @returns {Object} An object with layer information or undefined if it cannot be found
```

Comment:
Shouldn't it be a HTTP 404 in case it cannot be found?

Reply:
Returning undefined is done intentionally if someone wants to check if a layer a exists or not. The same behavior is done in `getFeatureType`, `getCoverage` and `getWmsLayer`. The function only throws an error if GeoServer cannot be reached at all. 
Is this okay for you?

Reply:
OK for consistency